src/timer/TimingInterface.ts

```diff
@@ -0,0 +1,148 @@
+class TimingInterface {
+    initialTime: number;
+    currentTime: number;
+    rate:number;
+    timerId: number;
+
+    constructor (duration: number, rate: number = 1) {
+        this.initialTime = duration;
+        this.currentTime = duration;
+        this.rate = rate;
+    }
+
+    getCurrentTime() {
+        return this.currentTime;
+    }
+
+    setCurrentTime(duration: number) {
+        this.currentTime = duration;
+    }
+
+    getRate() {
+        return this.rate;
+    }
+    
+    setRate(rate: number) {
+        this.rate = rate;
+    }
+///////////////////////////////////////////////////////////////
+    isZero() {
+        return Math.floor(this.getCurrentTime()) == 0;
+    }
+
+    isPositive() {
+        return Math.floor(this.getCurrentTime()) > 0;
+    }
+
+    addTime(duration: number) {
+        this.currentTime += duration;
+    }
+
+    subtractTime(duration: number) {
+        this.currentTime -= duration;
+    }
+
+    increment() {
+        this.addTime(1);
+    }
+
+    decrement() {
+        this.subtractTime(1);
+    }
+
+    getMinutes() {
+        return Math.floor(this.getCurrentTime() / 60);
+    }
+
+    getSeconds() {
+        return Math.floor(this.getCurrentTime() % 60);
+    }
+///////////////////////////////////////////////////////////////////////////////
+    startTime() {
```

Comment:
I think it would be better to rename these functions. Remove time from the name as we already know this is timer.
If we have for example:
`const timer = new TimingInterface(...)`
it is more intuitive to call 
`timer.start or timer.pause`
as opposed to
`timer.startTime`